client/kubeconfig.go

```diff
@@ -3,44 +3,82 @@
 
 package client
 
+//go:generate ../bin/mockgen -source=kubeconfig.go -copyright_file=../hack/copyright_header.txt -destination=pkg/mocks/mock_kube.go -package=mocks github.com/Azure/aks-secure-tls-bootstrap/client KubeClient
+
 import (
 	"fmt"
 	"os"
 	"time"
 
 	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/transport"
 	certutil "k8s.io/client-go/util/cert"
 )
 
-func isKubeConfigStillValid(kubeConfigPath string, logger *zap.Logger) (bool, error) {
-	logger.Debug("checking if kubeconfig exists...")
+type KubeClient interface {
+	IsKubeConfigStillValid(kubeConfigPath string) (bool, error)
+	EnsureClusterConnectivity(kubeConfigPath string) error
```

Comment:
I'm actually thinking the better name for this method would be `EnsureAuthentication` or similar